org.adempiere.base/src/org/compiere/print/DataEngine.java

```diff
@@ -1042,26 +1044,51 @@ else if (m_group.isFunctionColumn(pdc.getAD_PrintFormatItem_ID(), functions[f]))
 							}
 							else
 							{
+								
+
+								
 								String id = rs.getString(counter++);
 								if (display != null && !rs.wasNull())
 								{
 									/** START DEVCOFFEE: script column **/
 									int displayType = pdc.getDisplayType();
-									if (display.startsWith("@SCRIPT"))
-									{
-										displayType = DisplayType.Number;
-										display = display.replace("@SCRIPT", "");
-										display = parseVariable(display, pdc, pd);
-										Interpreter bsh = new Interpreter ();
-										try {
-											display = bsh.eval(display).toString();
-										} catch (EvalError e) {
-											log.severe(e.getMessage());
+									if(rs.getMetaData().getColumnName(displayIndex).contains("SCRIPTCOLUMN")) {
+										Object value = rs.getObject(displayIndex);
+										int columnType = rs.getMetaData().getColumnType(displayIndex);
+										// String, Text
+										if (columnType == Types.CHAR || columnType == Types.VARCHAR)
+										{
+											displayType = DisplayType.Text;
+										}
+										else if (columnType == Types.DATE || columnType == Types.TIME
+												|| columnType == Types.TIMESTAMP)
+										{
+											displayType = DisplayType.DateTime;
+										}
+										// Number
+										else if (columnType == Types.INTEGER || columnType == Types.SMALLINT
+											|| columnType == Types.DECIMAL || columnType == Types.NUMERIC)
+										{
+											displayType = DisplayType.Number;
 										}
-									}
 
-									ValueNamePair pp = new ValueNamePair(id, display);
-									pde = new PrintDataElement(pdc.getAD_PrintFormatItem_ID(), pdc.getColumnName(), pp, displayType, pdc.getFormatPattern());
+										if (display.startsWith("@SCRIPT"))
+										{
+											displayType = DisplayType.Number;
+											display = display.replace("@SCRIPT", "");
+											value = parseVariable(display, pdc, pd);
+											Interpreter bsh = new Interpreter ();
+											try {
+												display = bsh.eval(display).toString();
+											} catch (EvalError e) {
+												log.severe(e.getMessage());
+											}
+										}
+										pde = new PrintDataElement(pdc.getAD_PrintFormatItem_ID(), pdc.getColumnName(), (Serializable) value, displayType, pdc.getFormatPattern());
```

Comment:
This is incomplete (and hence inconsistent), should handle the code path for "@SCRIPT" too. Similar to the db code path, a couple of instanceof check should be sufficient to determine the return type from bsh.eval(display).

Reply:
Comited fix for eval function, missed change of variable. Path for @SCRIPT is limited by method parseVariable which Can get only use BigDecimal Columns atm and return only numeric String to evaluation so it will be always numeric.